keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -795,79 +846,192 @@ enhancement:
 
+This depends on the workload and CSI driver. A very low rate of
+provisioned volumes per second and few total number of volumes overall
+may be fine for some workloads (long-running applications which use
+specialize storage like PMEM) while others may need a much higher rate
+(adding a local LVM scratch volume to every pod in the system).
+
+The goal is to achieve the same provisioning rates with the feature
+enabled as those that currently can be achieved without it.
+
+This will need further discussion before going to GA.
+
 * **Are there any missing metrics that would be useful to have to improve
   observability if this feature?**
 
-### Dependencies
+No.
 
-Will be added before the transition to beta.
+### Dependencies
 
 * **Does this feature depend on any specific services running in the cluster?**
 
-### Scalability
+For core Kubernetes just the ones that will also run without it enabled (apiserver,
+kube-scheduler). Additional services are the CSI drivers.
+
+ * CSI driver
+   * Usage description:
+     * Impact of its outage on the feature: pods that use the CSI driver will not
+       be able to start
+     * Impact of its degraded performance or high-error rates on the
+       feature: When storage capacity information is not updated or
+       not updated often enough, then pods are either not getting
+       scheduled in cases where they could be scheduled (free capacity
+       not reported) or they get tentatively scheduled onto nodes
+       which do not have enough capacity (exhausted capacity not
+       reported). To recover from the first scenario, the driver eventually
+       needs to report capacity. To recover from the second scenario,
+       volume creation attempts will fail with "resource exhausted" and
+       other nodes have to be tried.
 
-Will be added before the transition to beta.
+### Scalability
 
 * **Will enabling / using this feature result in any new API calls?**
 
+Yes.
+
+Enabling it in apiserver and CSI drivers will cause
+`CSIStorageCapacity` objects to be created or updated. The
+number of those objects is proportional to the number of storage
+classes and number of distinct storage topology segments. For
+centralized provisioning, the number of segments is probably low. For
+distributed provisioning, the each node where the driver runs
+represents one segment, so the total number is total number of objects
+is equal to the product of "number of nodes" and "number of storage
+classes".
+
+The rate at which objects depends on how often topology and storage
+usage changes. It can estimated as:
+* creating objects for each new node and deleting them when removing a
+  node when using distributed provisioning
+* the same for adding or removing storage classes (both modes)
+* updates when volumes are created/resized/deleted (thus bounded by
+  some other API calls)
+* updates when capacity in the underlying storage system is changed
```

Comment:
or when a PV gets provisioned

Reply:
That's covered by "when volumes are created".

Reply:
I don't see "when volumes are created" in this list. Can we be more explicit here?

Reply:
It's mentioned above:
"when volumes are created/resized/deleted" 

:)

Reply:
It's mentioned above:
"when volumes are created/resized/deleted" 

:)

Reply:
sorry I need new glasses :)